Refresh environments on initialize

dorkster100 opened the pull request:
@adalessa It worked for me, but when i reopened the editor - it stopped.

I once again did some debugging and found out, that it only works for me if I start editor in the project directory
```
nvim ~/projects/example
```

If i start the editor in a random dir (as I normally do tbh), docker-compose container_name is already set to ''.

The  reason is simple, we are not initializing environments again on DirChanged autocommand and the utils.get_env "APP_SERVICE"  is not resetting for new dir.

adalessa commented:
I see the point since is calculated and will be store. Another problem that I see is that it will require to restart your session if you just write the .env file. For a better user experience should be able to pick this on the fly. 
For now I think is good, could you update the readme to reflect that the environment configuration should return a function ?

dorkster100 commented:
> I see the point since is calculated and will be store. Another problem that I see is that it will require to restart your session if you just write the .env file. For a better user experience should be able to pick this on the fly. For now I think is good, could you update the readme to reflect that the environment configuration should return a function ?

I fixed readme wording if you don`t mind, however it turned out that there were no affected parts of the readme, unless we want to provide examples of creating custom environments.

I also found out that after changing from one laravel project to another it would not allow second initialization, so i moved some logic to `warmup` function.

Regarding .env file editing, i agree, it should, but maybe instead of doing autocommand and watching for `.env` file changes, we should just make `app.warmup` function available through :Laravel command?
